### DISCUSSION TITLE
Bit depth getting reduced from 16 to 8 on TIF images generated by DeepSkyStacker

### DISCUSSION THREAD

**Discussion by howlev** [ImageMagick/ImageMagick#4220] [Help]
Hi,

I am using Imagemagick for astronomical imaging processing.
I am trying to process depth 16 bit images from an astronomy program called DeepSkyStacker.

I am having issues with the current version of Imagemagick, where it is reducing the
bit depth of the images from  **depth 16-bit to  depth 16/8-bit**

( older versions of Imagemagick, and other programs like current Gimp are handling the images correctly)

This ONLY affects images created by DeepSkyStacker ( 16bit images created by other programs
are handled correctly by current versions of Imagemagick )

convert.exe (the old version) Version: ImageMagick 6.8.9-0 Q16 x86 2014-04-06
convert.exe (the new version) Version: ImageMagick 7.1.0-5 Q16 x86 2021-08-22

I have placed a sample image dss.tif  ( a grayscale depth 16 image ) on drop box

[https://www.dropbox.com/s/rry1b3rdug529t3/dss.tif?dl=0](url)

running 'convert.exe dss.tif -verbose info:'

 Old convert produces  
    .......................
    Type: Grayscale
    Base type: Grayscale
    Endianess: Undefined
    Colorspace: Gray
    **Depth: 16-bit**
    Channel depth:
      gray: 16-bit
    Channel statistics:
      Gray:
        min: 2987 (0.0455787)
        max: 65535 (1)
        mean: 5198.31 (0.0793211)
        standard deviation: 1331.21 (0.020313)
        kurtosis: 566.681
        skewness: 14.3413
    ...........................................................

New convert produces  
    ................
    Colorspace: Gray
    Type: Grayscale
    Endianness: Undefined
    **Depth: 16/8-bit**
    Channel depth:
      Gray: 8-bit
    Channel statistics:
      Pixels: 10762752
      Gray:
        min: 2827  (0.0431373)
        max: 65535 (1)
        mean: 5090.86 (0.0776816)
        median: 4112 (0.0627451)
        standard deviation: 1337.42 (0.0204078)
        kurtosis: 561.387
        skewness: 14.2497
        entropy: 0.494935
    Colors: 245
    .................................................

   When i run a dump of the histogram data using  'convert.exe  dss.tif -format '%c' histogram:info:'

       The old convert generates 3092 values

       The new convert generates only 245 values

   Is there anyway way that I can use the current version of ImageMagick on these
   images and still maintain the bit depth ?

   Thanks,
   Howie

**Comment by fmw42** [comment]
When your run convert.exe you will get IM 6 behavior. Perhaps your IM 6 is Q8?  Try 

`magick.exe dss.tif -verbose info:`

When I use

`magick dss.tif -verbose info:`

on IM 7.1.0.7 or 7.1.0.5 Q16 Mac OSX, I get 16-bit depth, not 16/8

```
 Filename: dss.tif
  Format: TIFF (Tagged Image File Format)
  Mime type: image/tiff
  Class: DirectClass
  Geometry: 3432x3136+0+0
  Resolution: 100x100
  Print size: 34.32x31.36
  Units: PixelsPerInch
  Colorspace: Gray
  Type: Grayscale
  Endianness: Undefined
  Depth: 16-bit
  Channel depth:
    Gray: 16-bit

```

Perhaps it is a Windows only issue?

**Comment by dlemstra** [comment]
Adding image to make sure it is preserved: [dss.zip](https://github.com/ImageMagick/ImageMagick/files/7159249/dss.zip)

**Comment by snibgo** [comment]
IM v6.9.9-40 on Windows 8.1 reports a problem:
```
f:\web\im>%IMDEV%convert dss.tif info:
convert: improper image header `dss.tif' @ error/tiff.c/ReadTIFFImage/1597.
convert: no images defined `info:' @ error/convert.c/ConvertImageCommand/3258.

f:\web\im>%IMDEV%convert -version
Version: ImageMagick 6.9.9-40 Q32 x86_64 2021-02-23 http://www.imagemagick.org
Copyright: © 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP
Delegates (built-in): bzlib cairo fftw fontconfig fpx freetype jbig jng jpeg
lcms ltdl lzma pangocairo png rsvg tiff webp wmf x xml zlib
```

**Reply by howlev** [reply]
I am running it from windows,   and get the  issue with both windows7 and windows10

**Reply by fmw42** [reply]
What do you get from 

`convert -version `

Are you on Q8 or Q16 compile?

**Reply by howlev** [reply]
I'm getting the following (  looks like  Q16  compile)

Version: ImageMagick 7.1.0-5 Q16 x86 2021-08-22 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Visual C++: 192930133
Features: Cipher DPC OpenCL
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib

**Reply by fmw42** [reply]
I do not understand why you would get 16/8 for your image. When I do that on Mac ImageMagick, I get 16. So perhaps a Windows issue will your version. One of the IM developers will need to comment further.

**Reply by dlemstra** [reply]
It looks like this is fixed in that latest release.
